modules/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureBlobStore.java

```diff
@@ -720,11 +732,49 @@ Map<String, Long> statsMap(boolean stateless) {
             }
         }
 
-        public void onSuccessfulRequest(Operation operation, OperationPurpose purpose) {
-            collectors.computeIfAbsent(new StatsKey(operation, purpose), k -> new LongAdder()).increment();
+        public void onRequestComplete(Operation operation, OperationPurpose purpose, AzureClientProvider.RequestMetrics requestMetrics) {
+            final StatsKey statsKey = new StatsKey(operation, purpose);
+            final LongAdder counter = opsCounters.computeIfAbsent(statsKey, k -> new LongAdder());
+            final Map<String, Object> attributes = opsAttributes.computeIfAbsent(
+                statsKey,
+                k -> RepositoriesMetrics.createAttributesMap(repositoryMetadata, purpose, operation.getKey())
+            );
+
+            counter.add(requestMetrics.getRequestCount());
```

Comment:
I wonder whether we should split this change into a separate PR. This is a slight change in behaviour so feels better to be in its own PR (albeit tiny). Also, I remembered this comment https://github.com/elastic/elasticsearch/pull/104444#discussion_r1455068279 which could use some clarification